// teboho-slider/fesm2015/teboho-slider.mjs
import * as i0 from '@angular/core';
import { Component, Input, HostListener, NgModule } from '@angular/core';
import * as i1 from '@angular/common';
import { CommonModule } from '@angular/common';

class TebohoSliderComponent {
    constructor() {
        this.data = [];
        this.cardContainers = window.innerWidth > 990 ? 5 : 3;
        this.containerArrayBlock = new Array(this.cardContainers);
        this.sliderData = [];
        this.selectedIndex = 0;
    }
    onResize(event) {
        let windowWidth = event.target.innerWidth;
        if (windowWidth > 990) {
            this.containerArrayBlock = new Array(5);
        }
        else {
            this.containerArrayBlock = new Array(3);
        }
    }
    ngOnInit() {
        this.setSliderItems();
        this.onAutoScroll();
    }
    getClass(i) {
        let className = '';
        if (window.innerWidth > 990) {
            switch (i) {
                case 0:
                case 4:
                    className = 'last';
                    break;
                case 1:
                case 3:
                    className = 'next';
                    break;
                case 2:
                    className = 'active';
                    break;
            }
        }
        else {
            switch (i) {
                case 0:
                case 2:
                    className = 'next';
                    break;
                case 1:
                    className = 'active';
                    break;
            }
        }
        return className;
    }
    onNext() {
        if (this.selectedIndex < this.data.length - 1) {
            this.selectedIndex++;
        }
        else {
            this.selectedIndex = 0;
        }
        this.setSliderItems();
    }
    onPrevious() { }
    setSliderItems() {
        let temp = 0;
        this.sliderData = [];
        for (let i = 0; i < this.cardContainers; i++) {
            if (this.data[i + this.selectedIndex]) {
                this.sliderData.push(this.data[i + this.selectedIndex]);
            }
            else {
                this.sliderData.push(this.data[temp]);
                temp++;
            }
        }
    }
    onAutoScroll() {
        if (Number(this.timer)) {
            setInterval(() => this.onNext(), Number(this.timer) * 1000);
        }
    }
}
TebohoSliderComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.11", ngImport: i0, type: TebohoSliderComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
TebohoSliderComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.2.11", type: TebohoSliderComponent, selector: "teboho-slider", inputs: { data: "data", timer: "timer" }, host: { listeners: { "window:resize": "onResize($event)" } }, ngImport: i0, template: "<div class=\"slider\">\n  <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"26\" height=\"26\" fill=\"currentColor\"\n    class=\"bi bi-arrow-left-circle-fill previousButton\" viewBox=\"0 0 16 16\" (click)=\"onPrevious()\">\n    <path\n      d=\"M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0zm3.5 7.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z\" />\n  </svg>\n\n  <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"26\" height=\"26\" fill=\"currentColor\"\n    class=\"bi bi-arrow-right-circle-fill nextButton\" viewBox=\"0 0 16 16\" (click)=\"onNext()\">\n    <path\n      d=\"M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z\" />\n  </svg>\n\n  <div class=\"slider-container\">\n    <div class=\"card\" *ngFor=\"let item of containerArrayBlock; let i = index\" ngClass=\"{{ getClass(i) }}\">\n      <div class=\"title\">{{ sliderData[i].title }}</div>\n      <img [src]=\"sliderData[i].imageUrl\" class=\"image\" alt=\"\" />\n    </div>\n  </div>\n</div>\n", styles: [".slider{display:flex;flex-direction:column;position:relative}.slider .slider-container{width:100vw;height:28rem;overflow:hidden;background-color:#ffffff86;display:flex;justify-content:center;align-items:center;gap:1rem;font-family:sans-serif;position:relative}.slider .slider-container .card{width:16rem;height:14rem;border-radius:1rem;background-color:#fff;overflow:hidden;position:relative;box-shadow:0 6px 14px -4px #000000bf}.slider .slider-container .card .image{width:100%;height:100%;object-fit:cover}.slider .slider-container .card .title{position:absolute;background-color:#0008;color:#c7c7c7;bottom:0;width:100%;display:flex;justify-content:center;padding:.3rem}.slider .slider-container .card.active{transform:scale(1.8);z-index:10}.slider .slider-container .card.next{transform:scale(1.3);z-index:8}.slider .slider-container .card.last{transform:scale(1);z-index:6}.slider .previousButton,.slider .nextButton{position:absolute;bottom:50%;z-index:99;cursor:pointer;transition:.5s ease-in-out;font-size:5rem;fill:#fff}.slider .previousButton:hover,.slider .nextButton:hover{transform:scale(1.5);box-shadow:0 6px 14px -4px #000000bf}.slider .previousButton{left:2rem}.slider .nextButton{right:2rem}\n"], dependencies: [{ kind: "directive", type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i1.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.11", ngImport: i0, type: TebohoSliderComponent, decorators: [{
            type: Component,
            args: [{ selector: 'teboho-slider', template: "<div class=\"slider\">\n  <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"26\" height=\"26\" fill=\"currentColor\"\n    class=\"bi bi-arrow-left-circle-fill previousButton\" viewBox=\"0 0 16 16\" (click)=\"onPrevious()\">\n    <path\n      d=\"M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0zm3.5 7.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z\" />\n  </svg>\n\n  <svg xmlns=\"http://www.w3.org/2000/svg\" width=\"26\" height=\"26\" fill=\"currentColor\"\n    class=\"bi bi-arrow-right-circle-fill nextButton\" viewBox=\"0 0 16 16\" (click)=\"onNext()\">\n    <path\n      d=\"M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z\" />\n  </svg>\n\n  <div class=\"slider-container\">\n    <div class=\"card\" *ngFor=\"let item of containerArrayBlock; let i = index\" ngClass=\"{{ getClass(i) }}\">\n      <div class=\"title\">{{ sliderData[i].title }}</div>\n      <img [src]=\"sliderData[i].imageUrl\" class=\"image\" alt=\"\" />\n    </div>\n  </div>\n</div>\n", styles: [".slider{display:flex;flex-direction:column;position:relative}.slider .slider-container{width:100vw;height:28rem;overflow:hidden;background-color:#ffffff86;display:flex;justify-content:center;align-items:center;gap:1rem;font-family:sans-serif;position:relative}.slider .slider-container .card{width:16rem;height:14rem;border-radius:1rem;background-color:#fff;overflow:hidden;position:relative;box-shadow:0 6px 14px -4px #000000bf}.slider .slider-container .card .image{width:100%;height:100%;object-fit:cover}.slider .slider-container .card .title{position:absolute;background-color:#0008;color:#c7c7c7;bottom:0;width:100%;display:flex;justify-content:center;padding:.3rem}.slider .slider-container .card.active{transform:scale(1.8);z-index:10}.slider .slider-container .card.next{transform:scale(1.3);z-index:8}.slider .slider-container .card.last{transform:scale(1);z-index:6}.slider .previousButton,.slider .nextButton{position:absolute;bottom:50%;z-index:99;cursor:pointer;transition:.5s ease-in-out;font-size:5rem;fill:#fff}.slider .previousButton:hover,.slider .nextButton:hover{transform:scale(1.5);box-shadow:0 6px 14px -4px #000000bf}.slider .previousButton{left:2rem}.slider .nextButton{right:2rem}\n"] }]
        }], ctorParameters: function () { return []; }, propDecorators: { data: [{
                type: Input,
                args: ['data']
            }], timer: [{
                type: Input,
                args: ['timer']
            }], onResize: [{
                type: HostListener,
                args: ['window:resize', ['$event']]
            }] } });

class TebohoSliderModule {
}
TebohoSliderModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.11", ngImport: i0, type: TebohoSliderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TebohoSliderModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.11", ngImport: i0, type: TebohoSliderModule, declarations: [TebohoSliderComponent], imports: [CommonModule], exports: [TebohoSliderComponent] });
TebohoSliderModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.11", ngImport: i0, type: TebohoSliderModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.11", ngImport: i0, type: TebohoSliderModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        TebohoSliderComponent
                    ],
                    imports: [
                        CommonModule
                    ],
                    exports: [
                        TebohoSliderComponent
                    ]
                }]
        }] });

/*
 * Public API Surface of teboho-slider
 */

/**
 * Generated bundle index. Do not edit.
 */

export { TebohoSliderComponent, TebohoSliderModule };
