// teboho-slider/esm2020/lib/teboho-slider.module.mjs
import { NgModule } from '@angular/core';
import { TebohoSliderComponent } from './teboho-slider.component';
import { CommonModule } from "@angular/common";
import * as i0 from "@angular/core";
export class TebohoSliderModule {
}
TebohoSliderModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.2.11", ngImport: i0, type: TebohoSliderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TebohoSliderModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.2.11", ngImport: i0, type: TebohoSliderModule, declarations: [TebohoSliderComponent], imports: [CommonModule], exports: [TebohoSliderComponent] });
TebohoSliderModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.2.11", ngImport: i0, type: TebohoSliderModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.2.11", ngImport: i0, type: TebohoSliderModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        TebohoSliderComponent
                    ],
                    imports: [
                        CommonModule
                    ],
                    exports: [
                        TebohoSliderComponent
                    ]
                }]
        }] });
